DoSomething js i18n: wrap hardcoded strings into Drupal.t().

sergiitk opened the pull request:
#### What's this PR do?
- Wraps all Paraneue DoSomething hardcoded strings found in javascripts files into [`Drupal.t()`](https://www.drupal.org/node/323109) function
#### How should this be manually tested?
- Build javascript: `bin/ds grunt`
- Open [http://localhost:8888/](http://localhost:8888/)
- Run `drush cc css-js`
- Refresh [http://localhost:8888/](http://localhost:8888/)
- Check for one of srings (see diff) in the [translation interface](http://localhost:8888/admin/config/regional/translate/translate)

You can gain access to it by running `drush upwd admin --password 1` and then log in as `admin@example.com` with password `1`. 
#### What are the relevant tickets?

Step 1: #2878: wrap PHP templates text into t().
Jira: [#DOS-91](https://jira.dosomething.org/browse/DOS-91)
#### Known bugs:
1. <del>`bin/ds grunt --watch` multiple errors: `Drupal is not defined`</del>
2. `bin/ds grunt --watch` fails tests: `ReferenceError: Can't find variable: Drupal`

@DFurnes, it should be fixed when we include drupal.js to grunt, right?
@weerd Could you please review code?
CC: @mshmsh5000 @aaronschachter

Commit:
DoSomething js i18n: wrap hardcoded strings into Drupal.t().

Commit:
Fixes spelling.

blisteringherb commented on lib/themes/dosomething/paraneue_dosomething/js/validation/auth.js:
```diff
@@ -93,41 +93,41 @@ define(function(require) {
       return done({
         success: true,
-        message: "Yikes, you're old!"
+        message: Drupal.t("Yikes, you're old!")
```
@DFurnes Has a PR in now to change this text. https://github.com/DoSomething/dosomething/pull/2930

Commit:
Fixes #2930 conflict.

weerd committed:
Merge pull request #2932 from weerd/modal--number-validation-update

Updating number validation in reportbacks to not allow 0.

weerd commented on lib/themes/dosomething/paraneue_dosomething/js/finder/FormView.js:
```diff
@@ -115,7 +115,7 @@ define(function(require) {
       if (FormView.$div === null) {
         // Nope. I'm gonna vomit...
-        throw "Error: FormView is not initialized.";
+        throw Drupal.t("Error: FormView is not initialized.");
```
This isn't really a user facing error. Not sure if they _need_ to be translated, but I'm assuming it won't hurt if they are. Just pointing it out ;)

sergiitk replied:
Thank you! That's exactly the way I was thinking I was deciding if I should wrap it into `Drupal.t`.

weerd commented on lib/themes/dosomething/paraneue_dosomething/js/validation/reportback.js:
```diff
@@ -8,12 +8,12 @@ define(function (require) {
       return done({
         success: false,
-        message: "Enter a number!"
+        message: Drupal.t("Enter a number!")
```
@sergii-tkachenko I've changed this in my recent PR to read "Enter a valid number!" because now we're filtering out `0`, which is a number but not valid for the field since we're looking for something above `0`. You'll likely get a merge conflict from this. Just a heads up!

see: https://github.com/DoSomething/dosomething/blob/76e7533c0a8b7296b433f9f4de32675a13f417a4/lib/themes/dosomething/paraneue_dosomething/js/validation/reportback.js#L17

sergiitk replied:
Yes, thank you, already figured that. Fixing it at the moment.

weerd commented:
Visual review done. Looks good! :+1: :tropical_drink:

Commit:
Fixes #2932 conflict.

Conflicts:
	lib/themes/dosomething/paraneue_dosomething/js/validation/reportback.js
  https://github.com/DoSomething/dosomething/commit/a08513cdf33a76979806504e593eeaf8052ae41c#diff-f8e3e8adcb67a18db16bb238fd9ec714R17

Commit:
Add Drupal object to Grunt js globals.

Fixes issue "`bin/ds grunt --watch` multiple errors: `Drupal is not defined`".

sergiitk commented:
Known Issue 2 "`bin/ds grunt --watch` fails tests: `ReferenceError: Can't find variable: Drupal`" will be fixed by RequireJS refactoring @DFurnes  is currently working on.
It will include drupal.js into requisite js libraries.

I will reference for history here @DFurnes's PR when it's ready and ensure that tests play nice.

DFurnes commented:
Sweet! :+1:

sergiitk commented:
@weerd @DFurnes Thank you a lot, guys!
